src/rust/engine/serverset/src/retry.rs

```diff
@@ -82,12 +124,17 @@ mod tests {
       BackoffConfig::new(Duration::from_millis(1), 1.0, Duration::from_millis(1)).unwrap(),
       TimerHandle::default(),
     )
-    .unwrap();
-    assert_eq!(
-      Err(format!("Failed after 5 retries; last failure: bad")),
-      Retry(s)
-        .all_errors_immediately(|v: Result<u8, _>| v, 5)
-        .wait()
-    );
+      .unwrap();
+    let result = Retry(s)
+      .all_errors_immediately(
+        |v: Result<u8, _>| v,
+        RetryParameters::new(5, "identity".to_string())
+      )
+      .wait();
+    assert!(match result {
+      // Logs may have timestamps or other preceding information, but end with the error message.
+      Err(s) => s.ends_with("Failed after 5 retries for identity; last failure: bad"),
```

Comment:
This won't give a very clear assertion if it fails - did it fail because it was ok, or because the message was wrong?

How about something like:
```rust
let error = result.expect_err();
assert!(error.ends_with(...));
```

Of if you're feeling fancy, we could introduce [spectral](https://github.com/cfrancia/spectral) as a new test dep:
```rust
let error = result.expect_err();
assert_that(error).ends_with(...);
```